src/views/HomeView.vue

```diff
@@ -52,34 +56,37 @@
 </template>
 
 <script lang="ts" setup>
-import { ref, onMounted, watch } from 'vue'
-import { ElOption, ElSelect, ElSlider, ElButton } from 'element-plus'
+import { ref, onMounted } from 'vue'
+import { ElOption, ElSelect, ElSlider, ElButton, ElLoading } from 'element-plus'
 import { useRolesStore } from '@/stores/roles'
 import { useQuizStore } from '@/stores/quiz'
 import type { Role } from '@/domain/Role'
-import { ErrorMessage, SuccessMessage } from '@/services/messages'
+import { ErrorMessage } from '@/services/messages'
 
 const rolesStore = useRolesStore()
 const quizStore = useQuizStore()
-
+const loading = ref(false)
 const role = ref<Role | null>(null)
 const localSelectedRole = ref<string | null>(null)
 
 onMounted(async () => {
   await rolesStore.fetchRoles()
   if (rolesStore.roles.length > 0) {
     localSelectedRole.value = ''
-    handleRoleChange(rolesStore.roles[0])
   }
 })
 
-const handleRoleChange = (selectedRole: Role) => {
-  try {
-    role.value = selectedRole
-    quizStore.initializeQuizAnswers(selectedRole.knowledges)
-    quizStore.setSelectedRole(selectedRole)
-  } catch (error) {
-    ErrorMessage('Failed to set selected role')
+const handleRoleChange = (RoleName: string) => {
```

Comment:
```suggestion
const handleRoleChange = (roleName: string) => {
```
Variables are camelCase